internal/auth/authn/multiauth.go

```diff
@@ -0,0 +1,796 @@
+	defer m.wg.Done()
+
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	// Load immediately on start
+	if err := m.LoadAllAuthProviders(m.ctx); err != nil {
+		m.log.Warnf("Failed to load auth providers on startup: %v", err)
+	}
+
+	for {
+		select {
+		case <-ticker.C:
+			if err := m.LoadAllAuthProviders(m.ctx); err != nil {
+				m.log.Warnf("Failed to reload auth providers: %v", err)
+			}
+		case <-m.ctx.Done():
+			m.log.Info("Stopping auth provider loader")
+			return
+		}
+	}
+}
+
+// LoadAllAuthProviders reloads auth providers from the database with change detection
+func (m *MultiAuth) LoadAllAuthProviders(ctx context.Context) error {
+
+	// List all auth providers from database
+	providerList, status := m.authProviderService.ListAuthProviders(ctx, api.ListAuthProvidersParams{})
+	if status.Code != http.StatusOK {
+		return fmt.Errorf("failed to list auth providers: %v", status)
+	}
+
+	// Build map of provider keys from DB for tracking
+	dbProviderKeys := make(map[AuthProviderCacheKey]*api.AuthProvider)
+	if providerList != nil && len(providerList.Items) > 0 {
+		for i := range providerList.Items {
+			provider := &providerList.Items[i]
+			providerKey, err := m.getProviderKey(provider)
+			if err != nil {
+				m.log.Warnf("Failed to get key for auth provider %s: %v", lo.FromPtr(provider.Metadata.Name), err)
+				continue
+			}
+			dbProviderKeys[providerKey] = provider
+		}
+	}
+
+	m.dynamicProvidersMu.Lock()
+	defer m.dynamicProvidersMu.Unlock()
+
+	// Track which providers we've seen
+	processedKeys := make(map[AuthProviderCacheKey]bool)
+	addedCount := 0
+	updatedCount := 0
+	unchangedCount := 0
+
+	// Process all providers from DB
+	for providerKey, provider := range dbProviderKeys {
+		processedKeys[providerKey] = true
+
+		existingMiddleware, exists := m.dynamicProviders[providerKey]
+
+		if exists {
+			// Provider exists - check if it changed
+			changed, err := m.hasProviderChanged(existingMiddleware, provider)
+			if err != nil {
+				m.log.Warnf("Failed to check if provider %s changed: %v", lo.FromPtr(provider.Metadata.Name), err)
+				continue
+			}
+
+			if changed {
+				// Provider changed - reconstruct middleware
+				_, authMiddleware, err := m.createAuthMiddlewareFromProvider(provider)
+				if err != nil {
+					m.log.Warnf("Failed to update auth provider %s: %v", lo.FromPtr(provider.Metadata.Name), err)
+					continue
+				}
+				m.dynamicProviders[providerKey] = authMiddleware
+				m.log.Infof("Updated auth provider: %s", lo.FromPtr(provider.Metadata.Name))
+				updatedCount++
+			} else {
+				// Provider unchanged - keep existing
+				unchangedCount++
+			}
+		} else {
+			// New provider - create and add
+			_, authMiddleware, err := m.createAuthMiddlewareFromProvider(provider)
+			if err != nil {
+				m.log.Warnf("Failed to create auth provider %s: %v", lo.FromPtr(provider.Metadata.Name), err)
+				continue
+			}
+			m.dynamicProviders[providerKey] = authMiddleware
```

Comment:
would it be possible that this providerKey already exists? should it always be overwritten

Reply:
yes its possible , and it should be overwritten - it means it was updated , 
we should not have two different providers with the same key